### ISSUE TITLE
MKS LCD2004_ML support

### ISSUE THREAD

**Issue by bdn76** [KevinOConnor/klipper#1198]
What about MKS LCD2004_ML (two line, text based) support with MKS Gen_L V1.0

![dsc_3641](https://user-images.githubusercontent.com/17513453/52280909-f7f95f00-296d-11e9-983d-3b680ae9faa2.JPG)
![dsc_3642](https://user-images.githubusercontent.com/17513453/52280910-f7f95f00-296d-11e9-9c48-ad82027a6658.JPG)

?

**Comment by KevinOConnor** [comment]
That appears to be a standard "RepRapDiscount 2004 Smart Controller" display - see the section on that in the config/generic-ramps.cfg file.

-Kevin

**Comment by bdn76** [comment]
Ok, i'm try this later. In which case should the display print meaningful information?
1. Immediatly after start module.
2. Only after connect klippy to module and stop displaying after disconnect.
3. Only after start printing and stop displaying after printing end.

In other words, what is the degree of autonomy for displaying information?

**Comment by bdn76** [comment]
Sorry, but not work:
![dsc_3644](https://user-images.githubusercontent.com/17513453/52362284-44699b00-2a51-11e9-8508-c3e1596aa956.JPG)

**Comment by KevinOConnor** [comment]
In order for us to assist, you'll need to attach the Klipper log file - see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

**Comment by bdn76** [comment]
Sorry, no problem.
[log.zip](https://github.com/KevinOConnor/klipper/files/2844353/log.zip)

**Comment by KevinOConnor** [comment]
I don't know why the display did not work.  Generally (after a full power reset) the display wont show anything until it is initialized.  So, it is odd that it is showing some data.  I'd double check the wiring.  Otherwise I don't have any additional advice.

-Kevin

**Comment by bdn76** [comment]
I'm check wires - all wires are correct.

1. Can i send string to display via cli before starting print?
2. Can you modify MCU firmware for print something to display immediatly after firmware start (idealy is a version and time elapsed after start with 1 secod update). This is simplify wire problem issue.

**Comment by bensen-666** [comment]
Yes, Same Problem with my 2004 display.

**Comment by KevinOConnor** [comment]
Any further luck with this?

-Kevin

**Comment by bdn76** [comment]
No. I'm disconnect and power off disaplay.

**Comment by KevinOConnor** [comment]
Okay, I'm going to close this then.  It is not known why it is not working and it doesn't look like there are any leads to getting it to work.

-Kevin

**Comment by humbuckr** [comment]
I recently had my stock Ender 3 main board lose its ability to control the bed temperature.  Googling lead me to believe i needed a new main board.  I got the MKS Gen 1.4 and the TMC2208 Stepper drivers.  

All I see on the STOCK LCD is the **backlight** after repeated firmware updates.

Things I've tried - 

- I've flipped the ribbon cable around
(flipped around on the slot 3 of the LCD is the only location where it lights up the LCD
- commented out // #define CR10_STOCKDISPLAY in configuration.h AND
- //#define REPRAP_DISCOUNT_SMART_CONTROLLER (tried both on and off)
- Swapped out the MKS Gen 1.4 with the stock creality board and the LCD displays the Ender menus as normal

Under a deadline to prints so I'm frustrated.    Would appreciate any tips.  thanks
